main.go

```diff
@@ -170,7 +173,7 @@ func main() {
 
 	cableEngineSyncer.Run(stopCh)
 
-	publicIPWatcher := getPublicIPWatcher(&submSpec, k8sClient, submarinerClient, localEndpoint)
+	publicIPWatcher := getPublicIPWatcher(&submSpec, k8sClient, submarinerClient, localEndpoint, airGapped)
```

Comment:
Minor thing but instead of passing in the flag I think it makes more sense to just check it outside:

```suggestion
	var publicIPWatcher *endpoint.PublicIPWatcher
	if !airGapped {
	    publicIPWatcher = getPublicIPWatcher(&submSpec, k8sClient, submarinerClient, localEndpoint)
	 }
```

Reply:
This is leading to the following error from the linter.
```
main.go:92:1: cognitive complexity 31 of func `main` is high (> 30) (gocognit)
```

In order to address the error, I will have to re-factor this main() method and since its not part of this feature, it might be worth doing it in a separate PR. Hope you are okay.

Reply:
ok - I can refactor afterwards.